core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -588,6 +633,70 @@ class MasterSuite extends SparkFunSuite
     }
   }
 
+  test("SPARK-19900: there should be a corresponding driver for the app after relaunching driver") {
+    val conf = new SparkConf().set("spark.worker.timeout", "1")
+    val master = makeMaster(conf)
+    master.rpcEnv.setupEndpoint(Master.ENDPOINT_NAME, master)
+    eventually(timeout(10.seconds)) {
+      val masterState = master.self.askSync[MasterStateResponse](RequestMasterState)
```

Comment:
Hi, this can not be moved because `MasterStateResponse` is changed over time. If we move the rpc out, the masterState will never change, and the assert will fail.

See the above test `SPARK-20529:...`, there is a same eventually assert.